airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/requester.py

```diff
@@ -144,3 +144,11 @@ def send_request(
 
         If a log formatter is provided, it's used to log the performed request and response. If it's not provided, no logging is performed.
         """
+
+    @property
+    @abstractmethod
+    def max_retries(self) -> Optional[int]:
```

Comment:
How will this behave for other retrievers like the polling one? Will the retries apply on the sets of requests or for each request? 

Having this question makes me feel like the abstraction should be only on the requester level as retries might not be represented in the same way depending on the type of retrieval. For example, the retry for polling might look like:
```
{
  "job_creation": 10,
  "waiting_for_job": 5,
  "fetch_job": 100
}
```